### PR TITLE
Added TLS support for MQTT connections.

### PR DESCRIPTION
**Pull request by samuelBloch**
In order to use TLS with  an MQTT connection, the user has to provide the path to the certificate (.cert.pem), private key (.private.key) and trust store (.der) files under "Controller Api MQTT" in the [Apache Felix Web Console Configurator](http://localhost:8080/system/console/configMgr).

The functionality of this feature has been tested with an AWS MQTT test client.

### CODE REVIEW

File: io.openems.edge.controller.api.mqtt/src/io/openems/edge/controller/api/mqtt/Config.java [hunk 1]
```diff
@@ -31,6 +31,18 @@
 
 	@AttributeDefinition(name = "Uri", description = "The connection Uri to MQTT broker.")
 	String uri() default "tcp://localhost:1883";
+	
+	@AttributeDefinition(name = "Certificate", description = "The path to the client certificate")
+	String certPath();
```
Comment (sfeilmeier) [inline io.openems.edge.controller.api.mqtt/src/io/openems/edge/controller/api/mqtt/Config.java:36]: Would it be possible or feasible to set the certificates directly as text here? In productive OpenEMS Edge distributions (like FENECON FEMS) the user is not allowed to upload (text)files to the edge device. If it was a direct configuration setting, the user could set it via OpenEMS UI.

Comment (samuelBloch) [reply io.openems.edge.controller.api.mqtt/src/io/openems/edge/controller/api/mqtt/Config.java]: Thanks for the input. The private key and certificates can now be pasted as strings.

**Comment by sfeilmeier** [comment]
Thanks for updating! I fixed the build problems locally. Could you please give me write permissions on your branch?

By the way, this PR relates to https://github.com/OpenEMS/openems/pull/1575. Unfortunately back then I did not get feedback from community to finalize the implementation.

**Comment by scootermon** [comment]
> Thanks for updating! I fixed the build problems locally. Could you please give me write permissions on your branch?

@sfeilmeier, Unfortunately, because the fork sits in an organization and not in a user-owned repo GitHub doesn't allow the setting to be turned on. I invited you directly to the fork so you can push the changes.

**Comment by sfeilmeier** [comment]
@scootermon Thank you. I pushed changes. Could you have a quick look? From my side its ready to merge.

**Comment by samuelBloch** [comment]
> @scootermon Thank you. I pushed changes. Could you have a quick look? From my side its ready to merge.

Thanks for the review and your changes. For me, everything looks fine, connection to the MQTT test client works too.

**Pull request merged**